source/agora/consensus/data/Transaction.d

```diff
@@ -97,6 +110,10 @@ public struct Transaction
 
     public void deserialize (scope DeserializeDg dg) nothrow @safe
     {
+        ubyte type;
+        deserializePart(type, dg);
+        this.type = cast(TxType)type;
```

Comment:
This cast should not be here. Instead, I suggest writing an explicit deserialization function in `Deserialize.d`:

```D
/// Ditto
public void deserializePart (ref TxType record, scope DeserializeDg dg)
    nothrow @trusted
{
    import std.traits;
    OriginalType!TxType orig_val;
    deserializePart(orig_val, dg);
    record = *cast(TxType*)(orig_val);
}
```

Reply:
Thanks for suggest.

Reply:
Ah sorry, I see that's what you did initially. @AndrejMitrovic : The problem with the explicit deserialization overload is that it brings in dependency. With this approach, the `Serialization` module will depend on every type in the program.

Reply:
Well I just worry about this code segment being duplicated anywhere TxType is used. For example, in the UTXO set Michael is working on.

Reply:
Random idea: `public void deserializePart (T) (ref T TxType ...) if (is(T == enum)` ?

Reply:
```D
/// Enum support
public void deserializePart (T)(ref T record, scope DeserializeDg dg) nothrow @trusted
    if (is(T == enum)) 
{
    import std.traits;
    OriginalType!T orig_val;
    deserializePart(orig_val, dg);
    record = *cast(T*)(orig_val);
}
```

Reply:
> ```d
> /// Enum support
> public void deserializePart (T)(ref T record, scope DeserializeDg dg) nothrow @trusted
>     if (is(T == enum)) 
> {
>     import std.traits;
>     OriginalType!T orig_val;
>     deserializePart(orig_val, dg);
>     record = *cast(T*)(orig_val);
> }
> ```

Is this the best?

Reply:
That addresses both @AndrejMitrovic and my concerns, so I think so. Sorry for the back and forth.

Reply:
not depend on consensus module  :)